src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -870,7 +870,7 @@ void FancyZones::UpdateDragState(HWND window, require_write_lock) noexcept
     const bool windowElevated = IsProcessOfWindowElevated(window);

```

Comment:
Can we do this check after `meElevated`? Since it's only needed if `meElevated` == false.
Also if `warning_shown` or `is_cant_drag_elevated_warning_disabled` are true, we don't need either of the two.

Reply:
>Can we do this check after meElevated?
yes, we already do that, see `if` stmt.

Reply:
alright, looks like my assumption regarding how C++ behavior wasn't correct! I thought short-circuiting will happen even with a dedicated variable - but it's in fact not true!

https://gcc.godbolt.org/z/rSRURH
https://gcc.godbolt.org/z/6hx7Vn